File: Sowndariya_Assessment25/Pages/registerPage.ts
import type {Page, Locator} from '@playwright/test';
export class RegisterPage{
    readonly page: Page;
    readonly myaccount: Locator;
    readonly register: Locator;
    readonly firstname: Locator;
    readonly lastname: Locator;
    readonly email: Locator;
    readonly phone: Locator;
    readonly password: Locator;
    readonly confirmpass: Locator;
    readonly policy: Locator;
    readonly continue:Locator;
    readonly createaccount:Locator;

    constructor(page:Page){
        this.page=page;
        this.myaccount=page.locator('span').filter({ hasText: 'My Account' });
        this.register=page.getByRole('link', { name: 'Register' });
        this.firstname=page.getByRole('textbox', { name: 'First Name' });
        this.lastname=page.getByRole('textbox', { name: 'Last Name' });
        this.email=page.getByRole('textbox', { name: 'E-Mail' });
        this.phone=page.getByRole('textbox', { name: 'Telephone' });
        this.password=page.locator('#input-password');
        this.confirmpass=page.locator('#input-confirm');
        this.policy=page.locator('[name="agree"]');
        this.continue=page.locator('input.btn.btn-primary');
        this.createaccount=page.getByRole('heading', { name: 'Your Account Has Been Created!' });
    }

    async navigate(){
        await this.page.goto(process.env.BASE_URL!);
        {
            waitUntil:`domcontentloaded`
        }
    }

    async registerdetail(
        firstname:string,
        lastname:string,
        phone:string,
        email:string,
        password:string,
        confirmpass:string
    ){
        await this.myaccount.click();
        await this.register.click();
        await this.firstname.fill(firstname);
        await this.lastname.fill(lastname);
        await this.phone.fill(phone);
        await this.email.fill(email);
        await this.password.fill(password);
        await this.confirmpass.fill(confirmpass);
        await this.policy.check();
        await this.continue.click();
    }


}

File: Sowndariya_Assessment24/Pages/registerPage.py
from selenium.webdriver.common.by import By


class RegisterPage:

    def __init__(self, driver):
        self.driver = driver

    def my_account(self):
        return self.driver.find_element(By.XPATH,
            "//span[text()='My Account']"
        )

    def register_link(self):
        return self.driver.find_element(
            By.LINK_TEXT,
            "Register"
        )

    def first_name(self):
        return self.driver.find_element(
            By.ID,
            "input-firstname"
        )

    def last_name(self):
        return self.driver.find_element(
            By.ID,
            "input-lastname"
        )

    def email(self):
        return self.driver.find_element(
            By.ID,
            "input-email"
        )

    def telephone(self):
        return self.driver.find_element(
            By.ID,
            "input-telephone"
        )

    def password(self):
        return self.driver.find_element(
            By.ID,
            "input-password"
        )

    def confirm_password(self):
        return self.driver.find_element(
            By.ID,
            "input-confirm"
        )

    def privacy_policy(self):
        return self.driver.find_element(
            By.NAME,
            "agree"
        )

    def continue_btn(self):
        return self.driver.find_element(
            By.XPATH,
            "//input[@value='Continue']"
        )

    def success_message(self):
        return self.driver.find_element(
            By.XPATH,
            "//h1[text()='Your Account Has Been Created!']"
        )